scylla/tests/integration/retries.rs

```diff
@@ -164,3 +164,70 @@ async fn retries_occur() {
         Err(err) => panic!("{}", err),
     }
 }
+
+// See https://github.com/scylladb/scylla-rust-driver/issues/1085
+#[tokio::test]
+#[ntest::timeout(30000)]
+#[cfg(not(scylla_cloud_tests))]
+async fn speculative_execution_panic_regression_test() {
+    use scylla_proxy::RunningProxy;
+
+    setup_tracing();
+    let test = |proxy_uris: [String; 3], translation_map, mut running_proxy: RunningProxy| async move {
+        let se = SimpleSpeculativeExecutionPolicy {
+            max_retry_count: 2,
+            retry_interval: Duration::from_millis(1),
+        };
+        let profile = ExecutionProfile::builder()
+            .speculative_execution_policy(Some(Arc::new(se)))
+            .retry_policy(Box::new(FallthroughRetryPolicy));
```

Comment:
Technically, this is a profile builder, not a profile. Maybe let's `build()` here, not later?